### PR TITLE
[MODULAR] [READY] Charge Weaponry

### PR DESCRIPTION
**Pull request by ShadeAware**
## About The Pull Request
Adds Charge Weapons, specialized station researched and produced ballistics that require both their own ammunition and a supply of power in order to function; the returns, however, are quite substantial. All weapons and ammunition are produced at the Security Protolathe after being researched in the **"Advanced Pulse-Shot Weaponry"** node.

Charge weapons require a charged internal power cell in order to function. The internal cell can be charged at any functional weapon charging dock. However, be careful to avoid overuse and EMP pulses; both will drain your cell extremely quickly, and can lead to dire situations where you have all the ammo in the world but no way to fire it.
EXAMPLE1
EXAMPLE2

## Why It's Good For The Game
The station has long needed more variety in the weapons it can use and produce. While the dynamic of Lasers against ballistics has always been an interesting affair, you never really saw the convergence of the two outside of a select few instances. This brings that convergence to the limelight, bringing the hard hitting stopping power of ballistic weapons with the careful drawbacks of energy weaponry.

## Changelog
:cl:
Add: Charged Weaponry, Charged Weaponry Icons, and "Advanced Pulse-Shot Weaponry" technology node for Security.
Edit: Weapon Charging Docks have been partially modularized to allow charge weaponry to fire, and can be more easily added onto at a later time period.
/:cl:

### CODE REVIEW

**Comment by ShadeAware** [comment]
Of course theres conflicts out the fucking gate.

**Comment by ShadeAware** [comment]
### **BINGO**

File: modular_skyrat/code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -36,3 +36,11 @@
 /datum/techweb_node/botany/New()
 	design_ids += "prisonerbiogenerator"
 	. = ..()
+
+/datum/techweb_node/charge_weaponry
+	id = "charge_weaponry"
+	display_name = "Advanced Pulse-shot Weapons"
+	description = "Utilizing pulse-slung ballistic charge technology, we can produce advanced ballistic weaponry to suit our needs."
+	prereq_ids = list("weaponry", "adv_weaponry", "emp_adv", "electronic_weapons")
```
Comment (Trilbyspaceclone) [inline modular_skyrat/code/modules/research/techweb/all_nodes.dm:44]: weaponry is not needed other then flavour,
Adv_weaponry needs normal is why.
If its a flavour then then its a-ok, game wise its still the same point locked wise

**Comment by Useroth** [comment]
I like this but not entirely sure about the balance if these are printable.

**Comment by Akrilla** [comment]
printable weapons are just a lategame nightmare

**Comment by Akrilla** [comment]
After a talk with the other maintainers I'm not sure if this is something we want. Biggest issue is just that they can be printed - it's just a bit too much. Closing this PR but can be reopened after speaking to us with ideas on improvements.

**Pull request closed without merging**